presto-cli/src/main/java/com/facebook/presto/cli/Query.java

```diff
@@ -198,11 +198,9 @@ private void pageOutput(OutputFormat format, List<String> fieldNames)
             if (!pager.isNullPager()) {
                 // ignore the user pressing ctrl-C while in the pager
                 ignoreUserInterrupt.set(true);
-                Thread clientThread = Thread.currentThread();
                 pager.getFinishFuture().thenRun(() -> {
                     userAbortedQuery.set(true);
                     ignoreUserInterrupt.set(false);
-                    clientThread.interrupt();
```

Comment:
We actually want to interrupt `processRows()`, but not anything else. So I think it should do something like this:
```java
AtomicBoolean processing = new AtomicBoolean(true);
pager.getFinishFuture().thenRun(() -> {
    ...
    synchronized (processing) {
        if (processing.get()) {
            clientThread.interrupt();
        }
    }
}
handle.processRows(client);
synchronized (processing) {
    processing.set(false);
}
```
This is ugly, but I'm not sure we can do anything better without fundamentally changing `StatementClient` to be async (and possibly other things too).